server/models/Collective.js

```diff
@@ -2136,31 +2136,42 @@ export default function (Sequelize, DataTypes) {
 
   Collective.prototype.getBalance = async function (until) {
     until = until || new Date();
-    const transactions = await models.Transaction.findOne({
-      attributes: [
-        [Sequelize.fn('COALESCE', Sequelize.fn('SUM', Sequelize.col('netAmountInCollectiveCurrency')), 0), 'total'],
-      ],
-      where: {
-        CollectiveId: this.id,
-        createdAt: { [Op.lt]: until },
-      },
-      raw: true,
-    });
-    const transactionSum = parseInt(transactions.total, 10);
+    const result = await this.sequelize.query(
+      `
+        WITH "blockedFunds" AS (
```

Comment:
Could we have this query in a different file to use the exact same one between the loader and the model?

Reply:
We can, but since `lib/queries` require models we have a cyclical dependency to refactor.
I'll try to do this right in this PR and see if we can DRY this query.